core/tests/puppeteer-acceptance-tests/user-utilities/translation-admin-utils.js

```diff
@@ -0,0 +1,177 @@
+   'button#add-contribution-rights-submit-button';
+
+// "Remove Contribution Rights" form elements.
+const revokeContributorUsernameInput =
+  'input#revoke-contribution-rights-user-input';
+const revokeContributonRightsCategorySelect =
+  'select#revoke-contribution-rights-category-select';
+const revokeContributonRightsLanguageSelect =
+  'select#revoke-contribution-rights-language-select';
+const revokeContributionRightsSubmitButton =
+  'button#revoke-contribution-rights-submit-button';
+
+module.exports = class TranslationAdmin extends baseUser {
+  /**
+   * Function for navigating to the contributor dashboard admin page.
+   */
+  async navigateToContributorDashboardAdminPage() {
+    await this.goto(ContributorDashboardAdminUrl);
+  }
+
+  /**
+   * Function for assigning a translation right to a user.
+   * @param {string} username - the username of the user.
+   * @param {string} languageValue - the language to assign as an option value.
+   */
+  async assignTranslationRights(username, languageValue) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, translationRightValue);
+    await this.select(addContributonRightsLanguageDropdown, languageValue);
+    await this.clickOn(addontributionRightsSubmitButton);
+
+    // Wait a moment for changes to take effect.
+    await this.page.waitForTimeout(1000);
+  }
+
+  /**
+   * Function for revoking a translation right from a user.
+   * @param {string} username - the username of the user.
+   * @param {string} languageValue - the language to revoke as an option value.
+   */
+  async revokeTranslationRights(username, languageValue) {
+    await this.type(revokeContributorUsernameInput, username);
+    await this.select(
+      revokeContributonRightsCategorySelect, translationRightValue);
+    await this.select(revokeContributonRightsLanguageSelect, languageValue);
+    await this.clickOn(revokeContributionRightsSubmitButton);
+  }
+
+  /**
+   * Function to display contributon rights by user.
+   * @param {string} username - the username of the user to view.
+   */
+  async viewContributorRightsByUser(username) {
+    await this.select(viewContributorMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+  }
+
+  /**
+   * Function to display translation rights by language.
+   * @param {string} languageValue - the language option as an option value.
+   */
+  async viewContributorTranslationRightsByLanguage(languageValue) {
+    await this.select(viewContributorMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, translationRightValue);
+    await this.select(viewContributorLanguageSelect, languageValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    // Wait a moment for usernames to appear.
+    await this.page.waitForTimeout(1000);
```

Comment:
avoid using waitForTimeouts, sometimes in throttled network the component might take more than 1sec to load and will fail the tests and will make it more flaky!.. instead of timeouts, try using waitForSelector for a component class that loads up when the loading is finished. also use networkIdle with 0 indicating that the loading is complete with 0 network requests (refer the other user utility files to know the usage of networkIdle'0')

Reply:
Replaced with [waitForNetworkIdle](https://pptr.dev/api/puppeteer.page.waitfornetworkidle)